lib/erlef/jobs.ex

```diff
@@ -0,0 +1,354 @@
+defmodule Erlef.Jobs do
+  @moduledoc """
+  The Jobs context.
+  """
+
+  alias Ecto.Changeset
+  alias Erlef.Repo
+  alias __MODULE__.Error
+  alias Erlef.Jobs.Post
+  alias Erlef.Jobs.PostHistoryEntry
+  alias Erlef.Accounts.Member
+
+  import Ecto.Query
+
+  @max_posts_per_author 4
+
+  @type create_post_params :: %{
+          title: String.t(),
+          description: String.t(),
+          position_type: :permanent | :contractor,
+          city: String.t() | nil,
+          region: String.t() | nil,
+          country: String.t() | nil,
+          remote: boolean(),
+          website: URI.t(),
+          days_to_live: pos_integer() | nil
+        }
+
+  @type update_post_params :: %{
+          title: String.t(),
+          description: String.t(),
+          position_type: :permanent | :contractor,
+          city: String.t() | nil,
+          region: String.t() | nil,
+          country: String.t() | nil,
+          remote: boolean(),
+          website: URI.t()
+        }
+
+  @doc """
+  Retrieves all job posts which are approved and not expired.
+
+  Sposored posts are in front of the list.
+  """
+  @spec list_posts() :: [Post.t()]
```

Comment:
Haven't looked into this, but on the face of it, this looks like it (the dialyzer error) is because this function can return an empty list.